### PR TITLE
Project 2 Submission

### PR DESCRIPTION
**Pull request by priyanshpsalian**
## Contributors
- Priyansh Salian (A20585026 psalian1@hawk.iit.edu)
- Shruti Ramchandra Patil (A20564354 spatil80@hawk.iit.edu)
- Pavitra Sai Vegiraju (A20525304 pvegiraju@hawk.iit.edu)
- Mithila Reddy (A20542879 Msingireddy@hawk.iit.edu)

## Project Overview
Gradient Boosting is a machine learning technique for regression and classification tasks, it combines several weak learners into strong learners where each new model is trained to minimize the loss function such as mean squared error or cross-entropy of this previous model using gradient desent. This project implements a **Gradient Boosting Regressor** from scratch, relying only on Python and NumPy. It serves as a learning tool and a practical implementation of Gradient Boosting, avoiding the use of external machine learning libraries like scikit-learn.

### CODE REVIEW

**Comment by priyanshpsalian** [comment]
done
